### ISSUE TITLE
Android: Local Notification In-App-messages do not show up on Android 9+ (and sometimes 8+)

### ISSUE THREAD

**Issue by KarsonKimbrel** [Adobe-Marketing-Cloud/mobile-services#295]
As of Android 8 (It is enforced in 9+ unless the target SDK is 8+, in which case it is also enforced in 8+), notifications will not show up unless a notification channel id is provided to the Notification Builder. 
Android documentation of Notification Channels: https://developer.android.com/training/notify-user/channels .

Allowing the app to inform the Adobe library of what channel id to use would be a sufficient fix.
This issue is around/at line 117 of MessageNotificationHandler

**Comment by subhowmi** [comment]
Hi @KarsonKimbrel,

Thanks for reporting the issue. We are looking into addressing notification channel support for the Android SDK. More details will be posted soon.

Thanks,
**Sudeep Bhowmick**
*Mobile Engineer*
[Adobe Marketing Cloud](https://www.adobe.com/experience-cloud.html)

**Comment by appleseedexm** [comment]
Has this been implemented yet? Mobile Services itself has the option to set a channel ID, but we were not able to get any notifications for Android 9+.

**Comment by shalehaha** [comment]
@appleseedexm 
The original ticket here was for Local notification and we haven't provided the support for it yet. But I assume you are asking for Push notification, right? For Push, you need to set up the channel on the App side, and use the correct channel when triggering the push message. The SDK is not involved in this process. 
Please contact client care https://helpx.adobe.com/contact/enterprise-support.ec.html if you need further help. Thanks.

Jiabin

**Comment by appleseedexm** [comment]
@shalehaha 
Yes, this is about Push notification and we have setup the local channels in the app which also work with Firebase, but not with AMS. I just wasn't sure if the ChannelId from AMS should actually work. We've been in contact with client care for weeks already, thought I'd ask since this sounded quite similar and we're out of ideas.

**Comment by jiabingeng** [comment]
@appleseedexm 

I saw a ticket for company `Ra******en`, is that from you?

Another note worths to mention: 
According to https://firebase.google.com/docs/cloud-messaging/concept-options, there are two types of message: Notification and Data. By default, Firebase uses Notification type, and Android handles the display of the notification for you. But AMS uses Data type, which means when the app receives the data, it is responsible for processing data messages.

**Comment by appleseedexm** [comment]
No that's not us, but thanks a lot for the help. That info is good to know, not heard of that from AMS so far. We've got it to work more or less right now but nobody really knows why. Will look into the Data vs Notification topic.

**Comment by diegocunhaiws** [comment]
Hey guys, do we have an answer for this? I'm facing this issue with all versions 8.0+ for Android.

**Comment by rymorale** [comment]
HI @diegocunhaiws,

It's been some time since you commented and if the issue is no longer occurring, please disregard the rest of this message.

Could you give more information about your issue? Is it the same as the original issue of "Android: Local Notification In-App-Messages do not show up on Android 9+ (and sometimes 8+)"? Or is it a similar issue to appleseedexm's push message issue? Could I get more information on which SDK you are using (mobile services or the ACP SDK's) and where you were generating the message (mobile services or Campaign)?

Thanks,
Ryan Morales
Mobile Engineer
[Adobe Experience Cloud](https://www.adobe.com/experience-cloud.html)

**Comment by diegocunhaiws** [comment]
Hey @rymorale,

The issue I faced with Notification In-App-Messages is the same of the original in Android 8+.

I used the Adobe Panel of my client to send this messages to app (Android and iOS)

**Comment by rymorale** [comment]
Hi @diegocunhaiws,

Thanks for the info. I just had an additional question. Are you using Adobe Mobile Services or Adobe Campaign in crafting the messages?

Thanks,
Ryan

**Comment by rymorale** [comment]
Hi @diegocunhaiws,

Are you still seeing this issue? If so, could we get information regarding your messaging setup? Are you using Adobe Mobile Services or Adobe Campaign in crafting the messages?

Thanks,
Ryan Morales
Mobile Engineer
[Adobe Experience Cloud](https://www.adobe.com/experience-cloud.html)
